Common/IsolatorLimitResultProvider.cs

```diff
@@ -0,0 +1,137 @@
+            )
+        {
+            isolatorLimitProvider.Consume(scheduledEvent.Name, () => scheduledEvent.Scan(scanTimeUtc));
+        }
+
+        /// <summary>
+        /// Executes the provided code block and while the code block is running, continually consume from
+        /// the limit result provided one token each minute. This function allows the code to run for the
+        /// first full second without requesting additional time from the provider. Following that, every
+        /// minute an additional one minute will be requested from the provider.
+        /// </summary>
+        /// <remarks>
+        /// This method exists to support scheduled events, and as such, intercepts any errors raised via the
+        /// provided code and wraps them in a <see cref="ScheduledEventException"/>. If in the future this is
+        /// usable elsewhere, consider refactoring to handle the errors in a different fashion.
+        /// </remarks>
+        public static void Consume(
+            this IIsolatorLimitResultProvider isolatorLimitProvider,
+            string name,
+            Action code
+            )
+        {
+            Exception error = null;
+            var memoryFence = new object();
+            var codeFinished = new ManualResetEvent(false);
+
+            // execute the code in a separate task so we can track time and request more as needed
+            Task.Run(() =>
+            {
+                code();
+                codeFinished.Set();
+            }).ContinueWith(task =>
+            {
+                // use full locking semantics to guarantee the value is propagated between threads
+                lock (memoryFence)
+                {
+                    // in the event an error is raised, capture the exception
+                    error = task.Exception;
+                    codeFinished.Set();
+                }
+            }, TaskContinuationOptions.OnlyOnFaulted);
+
+            // permit up to one second w/out requesting additional time from the provider
+            if (!codeFinished.WaitOne(Second))
+            {
+                var count = 0;
+                do
+                {
+                    if (count % 60 != 0)
+                    {
+                        continue;
+                    }
+
+                    // on the first iteration and every minute following, request additional time
+                    Log.Trace($"IsolatorLimitResultProvider.ConsumeForScheduledEvent({name}): Requesting additional time. Elapsed minutes: {count / 60}");
+                    if (!isolatorLimitProvider.TryRequestAdditionalTime(minutes: 1))
+                    {
+                        error = new TimeoutException("The scheduled event exceeded the available amount of time for processing. " +
+                                                    $"Elapsed: {count/60.0:0.0} minutes. Scheduled Event: {name}");
+                        break;
+                    }
+
+                    count++;
```

Comment:
I think this `count++;` should be before the first `continue`, 🤔 else it will never continue increasing?